python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -174,8 +174,7 @@ def _attribute_filter__set__(self, name, value):
         if value != "" and value is not None:
             #   Do some checking of the filenames in a subprocess
             if name == 'lfn':
-                if self.namePattern == '':
-                    self.namePattern = os.path.basename(value)
+                self.namePattern = os.path.basename(value)
```

Comment:
From my understanding the pattern in namePattern (if there is one) should then have multiple subfiles 1 for each match in order to do anything sensible. Then the IGangaFile acts as a sort of list which can be iterated (in some way).
If there is an IGangaFile without a wildcard (or equivalent) in the outputfiles then this is directly updated with the relevant file info after a job finishes.

If we ever support wildcards for DiracFile in the inputfiles I assume this would effectively be the same thing that we would have 1 file with the globbable string and then a set of subfiles where the actual data is.